File: src/C#/CreationalPatterns/AbstractFactory/EraStyles/Products/RenaissanceShip.cs
using EraStyles.Abstraction;
using EraStyles.Constants;
using System;
using System.Collections.Generic;
using System.Text;

namespace EraStyles.Products
{
    public class RenaissanceShip : EraObject
    {
        public override void ShowDetails()
        {
            Console.WriteLine(string.Format(GlobalConstants.ShipMessage, "Renaissance"));
        }
    }
}

File: src/JavaScript/CreationalPatterns/AbstractFactory/EraStyles/products/renaissanceShip.js

const EraObject = require("../abstraction/eraObject");
const GlobalConstants = require("../constants/globalConstants");

class RenaissanceShip extends EraObject {
    showDetails() {
        console.log(GlobalConstants.ShipMessage.replace("{0}", "Renaissance"));
    }
}

module.exports = RenaissanceShip;